src/thor/triplegbuilder.cc

```diff
@@ -1345,7 +1346,7 @@ void TripLegBuilder::Build(
 
   // Set origin, any through locations, and destination. Origin and
   // destination are assumed to be breaks.
-  CopyLocations(trip_path, origin, through_loc, dest, path_begin, path_end);
+  CopyLocations(trip_path, origin, throughs, dest, path_begin, path_end);
   auto* tp_orig = trip_path.mutable_location(0);
   auto* tp_dest = trip_path.mutable_location(trip_path.location_size() - 1);
 
```

Comment:
I was going to write a comment here about did you test whether we need to handle the special case of a single edge route, but t hinking it over, it seems its not possible. the change we made in route_action means that wherever there is an "intermediate" location we will have at least 2 edges. which means we can never hit the trivial 1 edge case below here.

in any case i would very highly  advise to start making gurka tests which excorcise this code. it will be much faster to validate the code that way than by  changing stuff here and trying it out on real data.